changelog/next/bug-fixes/5028--xsv-parser-fix

```diff
@@ -0,0 +1,4 @@
+We fixed a crash that could happen when reading in data with a known schema,
+enabling merging mode and then getting unexpected structural data for a field.
+In particular this could happen when reading in XSV with known schema, where the
+data contained unexpected list separators.
```

Comment:
While accurate, I think this is too complicated to understand. I would suggest rewording it like this:

> We fixed a bug in the `read_xsv` and `parse_xsv` family of operators and functions that caused the parser to fail unexpectedly when the data contained a list (as specified through the list separator), the `schema_only` and `schema` options were used, and the data contained a list in a field that the schema specified not to be a list.